### ISSUE TITLE
Error: log writing failed. Attempt to unlock a mutex which is locked by another thread/fiber

### ISSUE THREAD

**Issue by rxa1031** [github-changelog-generator/github-changelog-generator#974]
Version 1.16.4

Opened Bash from within a folder where I want to use the bash script.

The bash script executes below command:
```
github_changelog_generator --user "$ORG" --project "$THIS_PROJECT" --no-verbose --token "$GITHUB_API_TOKEN" --future-release "$TAG" -o "$CHANGELOG_MD"
```

Observed error:
![image](https://user-images.githubusercontent.com/18160755/127439875-c85a5690-bae3-4f06-954a-84d6d15588fb.png)

Please help fix the issue.

Thanks,
@rxa1031

**Comment by rxa1031** [comment]
I have not had this issue when using version 1.13.2. But it is no longer supported.

**Comment by ioquatix** [comment]
What version of Ruby are you using?

**Comment by rxa1031** [comment]
The command
gem --version

reports version as 3.2.8

**Comment by olleolleolle** [comment]
@rxa1031 The output of `ruby --version` is even more interesting.

**Comment by rxa1031** [comment]
$ ruby --version
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x64-mingw32]

**Comment by rxa1031** [comment]
$ git --version
git version 2.31.1.windows.1

**Comment by ioquatix** [comment]
Please update Ruby to 3.0.2 if possible. I believe it will fix your problem.

**Comment by rxa1031** [comment]
Thanks, working on it

**Comment by rxa1031** [comment]
I had to take help from IT to install the latest Rub+Devki. The FortiEDR was blocking installation and updates (over TCPIP). The script now works fine.
I will come back if I observe the issue again.
Thanks for helping.

**Comment by ioquatix** [comment]
Awesome!

**Comment by ioquatix** [comment]
@olleolleolle FYI/reference there is a bug in the recursive mutex implementation in Ruby 3.0.0 and it was fixed in 3.0.1 see https://bugs.ruby-lang.org/issues/17827 for more details.
